concourse/replicator.py

```diff
@@ -156,20 +156,12 @@ def _render(self, definition_descriptor):
         if bg := effective_definition.get('background_image'):
             pipeline_metadata['background_image'] = bg
 
-        # determine pipeline name (if there is main-repo, append the configured branch name)
         for variant in pipeline_metadata.get('definition').variants():
-            # hack: take the first "main_repository" we find
             if not variant.has_main_repository():
-                continue
-            main_repo = variant.main_repository()
-            pipeline_metadata['pipeline_name'] = '-'.join(
```

Comment:
I know - this is not part of the semantical focus of this PR. but: how about implementing a `dataclass` for `pipeline_metadata`? should (hopefully..) not be too much effort, but would increase maintainability a bit